lib/widgets/pinned_post.dart

```diff
@@ -58,7 +58,21 @@ class PinnedPost extends StatelessWidget {
                                       ? 'placeHolderUrl'
                                       : pinnedPost[index].imageUrl!,
                               errorWidget: (context, url, error) {
-                                return const CircularProgressIndicator();
+                                return Center(
+                                  child: Container(
+                                    width: SizeConfig.safeBlockHorizontal! * 8,
+                                    // Adjust the width to change the size of the circular indicator
+                                    height: SizeConfig.safeBlockVertical! * 4,
+                                    // Adjust the height to change the size of the circular indicator
+                                    child: AspectRatio(
+                                      aspectRatio: 1.0,
+                                      // Maintain a 1:1 aspect ratio for a circular shape
+                                      child: CircularProgressIndicator(
+                                        strokeWidth: 2.0,
+                                      ),
+                                    ),
+                                  ),
+                                );
```

Comment:
I think just wraping the CircularProgressIndicator into Center will fix it.

Reply:
Yes, @noman2002 , the suggested modification achieves the same visual result, but i think it utilizes the AspectRatio widget to maintain a 1:1 aspect ratio for the circular shape, ensures consistent appearance across various screen sizes.
Wrapping the CircularProgressIndicator with AspectRatio allows for more  adjusting the aspect ratio of the circular indicator and also by removing the Container, you lose the ability to specify explicit dimensions for the CircularProgressIndicator. It will now size itself based on its intrinsic size and the constraints of its parent widget.

So i made changes to the simple code as you told , so that it will fit every screen size used.

I will just change the code , if you think this is not required... to this one below
return Center(
         child: CircularProgressIndicator(
                 strokeWidth: 2.0,
         ),
  }